src/terrain/services/filesystem/stat.clj

```diff
@@ -115,6 +128,9 @@
 
 (defn do-stat
   [{user :user} body]
-  (-> (data-raw/collect-stats user body)
-      :body
-      (json/decode true)))
+  (let [paths         (:paths body)
+        ids           (:ids body)
+        request-user  (if ids user (get-public-data-user user paths))]
```

Comment:
This is a little finicky, and I don't really mind if it stays as it currently is, but it might be worth adding an extra arity to `get-public-data-user` -- my thinking is, support `[user paths]` like you have, but also `[user paths ids]`, where if `ids` is empty, use the `[user paths]` code, otherwise throw the `not-authorized` error like you are. That way, there's not two different pathways for errors -- I think right now, if you provide private paths, it'll throw not authorized above, but if you pass IDs it will probably throw an error for the missing "user" query parameter in data-info? Maybe? We're transitioning away from using IDs, but I think it's still nicer to be consistent.

Probably to do this, I'd put the bulk of the code into the `[user paths ids]` arity, and have `[user paths]` arity just be `([user paths] (get-public-data-user user paths nil))`

In case "multi-arity function" is meaningless to you, or you need a refresher, since I know you write less clojure than some: http://clojure-doc.org/articles/language/functions.html#multi-arity-functions

Reply:
Cool - thanks for this!  I'll refactor 👍 
In Clojure, I still don't quite have a concept of what's finicky or not, so comments like this are helpful for me 😄